src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -397,18 +406,26 @@ private void SerializeTracesLoop()
         private readonly struct WorkItem
         {
             public readonly Span[] Trace;
-            public readonly Action Callback;
+            public readonly Action<object> Callback;
+            public readonly object State;
```

Comment:
You're making the struct bigger, with a parameter that is almost never used. You should make sure it's not slowing down the 99% case (= workitems with a trace but no callback)

Reply:
🤔 makes sense I'm passing the struct as value so the bigger, the slower. This is probably a case of CPU vs Memory, and we prefer performance... I'll revert this change.

Reply:
done in 40744c8